phpunit/LDAP/AuthLdapTest.php

```diff
@@ -2175,6 +2175,27 @@ public function testRuleRight()
             'users_id' => $users_id,
         ]);
         $this->assertCount(1, $gus);
+
+        // update criteria
+        $criteria = new \RuleCriteria();
+        $crit_id = $criteria->update([
+            'id'  => $crit_id,
+            'pattern'   => 'brazil7',
+        ]);
+
+        // Check the user got the entity/profiles unassigned
+        $this->login('brazil6', 'password', false);
+        $users_id = \User::getIdByName('brazil6');
+        $this->assertGreaterThan(0, $users_id);
+
+        // Check group not assigned
+        $gu = new Group_User();
+        $gus = $gu->find([
+            'groups_id' => $group_id,
+            'users_id' => $users_id,
+        ]);
+
+        $this->assertCount(0, $gus);
```

Comment:
Maybe you shoiuld create a dedicated function for group process 

and  test all cases (for example)

1. Add 2 groups (manually)
1. Add 4 groups (dynamically)
2. Remove 2 groups (dynamically) 
3. Add 1 group / and remove 1 group (dynamically)